index.Rmd

```diff
@@ -429,6 +435,144 @@ You can also browse the history of a repository from your Git hosting service, a
 
 That's it: now you know how to keep track of all your files, so you will never lose them again, and no longer have to worry about making backups or saving multiple versions. In addition, this is the way that free and open-source code development happens in actuality. Also, the exercises and assignments in the course will be delivered and submitted this way, so make sure you are familiar with the whole process!
 
+## Project structure
+
+Another beneficial tool for scripting work is maintaining a consistent project structure as it will make it easier for you to switch from one project to the other and immediately understand how things work. In most cases, the project structure is entirely up to you. In some cases, however, the structure may be somewhat mandated, for instance, if you wish to make an R package. To make it easier to make a package out of your code, we will follow the structure of R packages:
+
+* A `main.R` script at the root of the project. This script performs step by step the different operations of your project. It is the only non-generic part of your project (it contains paths, already set variables, etc). If you are making a package, this should instead be a demo and live as `demo/main.R`.
+* An `R/` subdirectory: This directory should contain the functions you have defined as part of your project. These functions should be as generic as possible and are *sourced* and called by the `main.R` script. Or, if it's a package, `library(packagename)` makes them all available without needing to use `source()` manually.
+* A `data/` subdirectory: This directory contains data sets of the project. Since Git is not as efficient with non-text files, and GitHub has storage limits, you should only put small data sets in that directory (<2-3 MB). These can be geopackages, small rasters, csv files, or R archives. R offers two types of archives to store the important variables of the environments, `.rda` and `.rds`.
```

Comment:
Should read "GitLab" instead of GitHub, and we can remove the part about different R archives.